File: 0198-house-robber/solution.java
class Solution {
    public int rob(int[] nums) {
        if(nums.length==0) return 0;
        if(nums.length==1) return nums[0];

        int prev=nums[0];
        int prev1=Math.max(nums[0],nums[1]);
        int cur=prev1;

        for(int i=2;i<nums.length;i++){
            cur=Math.max(prev1,prev+nums[i]);
            prev=prev1;
            prev1=cur;
        }
        return cur;
    }
}

File: 0198-house-robber/solution.py
class Solution:
    def rob(self, nums: List[int]) -> int:
#            n=len(nums)
#            ec=0
#            oc=0
#           for i in range(0,n):
 #               if i%2==0:
  #                  ec+=nums[i]
   #             else:
    #                oc+=nums[i]
     #       return max(ec,oc)
        p1,p2=0,0
        for i in nums:
            cur=max(p1,p2+i)
            p2,p1=p1,cur
        return cur
    


        
